src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -328,6 +332,21 @@
   private List<String> includeOnlyProperties = Collections.emptyList();
 
   /**
+   * The option can be used to tell the plugin how it should generate the formerly known property 'git.commit.id'. Due to some naming issues when exporting the properties as an json-object (https://github.com/ktoso/maven-git-commit-id-plugin/issues/122) we needed to change the export of the property from 'git.commit.id' to 'git.commit.id.full'.
+   * However, due to the fact that this is one of the major properties the plugin is exporting we just don't want to change the exporting mechanism and somehow throw the backwards compatibility away.
+   * That's the point where this switch comes into place!
+   * By default it is set to 'true' and will generate the formerly known property 'git.commit.id' as it was in the previous versions of the plugin. With keeping the switch set to 'true' the plugin will print a warning that using this switch set to 'true' is deprecated and may be removed in a future release. However keeping it to 'true' by default preserve backwards compatibility and allows to migrate to the new properties when it's convenient.
+   * If you set this switch to 'false' the plugin will export the formerly known property 'git.commit.id' as 'git.commit.id.full'.
+   *
+   * *Note*: Depending on your plugin configuration you obviously can choose the 'prefix' of your properties by setting it accordingly in the plugin's configuration. As a result this is therefore only an illustration what the switch means when the 'prefix' is set to it's default value.
+   *
+   * @parameter default-value="true"
+   * @since 2.2.0
+   */
+  @Deprecated
+  private boolean generateCommitIdOldFashioned;
```

Comment:
one of the modes seems to me like it could be called "flat", since it'a a flatt representation, and the other mode is "full", so maybe a configurable field instead of boolean flag, enabling different nesting modes?